### PR TITLE
Unique numbers in an array in c++

### PR DESCRIPTION
**Pull request by HariniJeyaraman**
**Fixes issue:**

#4437

**Changes:**

Displays the unique numbers in a given integer array in C++

### CODE REVIEW

**Comment by HariniJeyaraman** [comment]
@arnavb @AdiChat Please review the code written in C++.

**Comment by Ankit152** [comment]
assign me this issue i fix this as an participant of GSSoC 2020

File: Unique_Numbers/unique_number.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
+#include<iostream>
+#include<stdlib.h>
+#include<algorithm>
+using namespace std;
```
Comment (arnavb) [inline Unique_Numbers/unique_number.cpp]: Remove this and move your code to this directory: https://github.com/OpenGenus/cosmos/tree/master/code/unclassified/src/unique_number

File: Unique_Numbers/unique_number.cpp [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+void display_unique(int arr[], int n)
+{
+    sort(arr, arr + n);
+    for(int i = 0; i < n; i++)
+    {
+        while(i < n-1 && arr[i] == arr[i+1])
```
Comment (arnavb) [inline Unique_Numbers/unique_number.cpp]: Reformat your code using clang-format.

File: Unique_Numbers/unique_number.cpp [hunk 3]
```diff
@@ -0,0 +1,40 @@
+#include<iostream>
+#include<stdlib.h>
+#include<algorithm>
+using namespace std;
+
+void display_unique(int arr[], int n)
+{
+    sort(arr, arr + n);
+    for(int i = 0; i < n; i++)
```
Comment (arnavb) [inline Unique_Numbers/unique_number.cpp]: Use pre-increment.

File: Unique_Numbers/unique_number.cpp [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+}
+
+int main()
+{
+    int n;
+    cout << "Enter the size of array : \n";
+    cin >> n;
+    int arr[n];
+    cout << "Enter the values for the array : \n";
+    for(int i = 0; i < n; i++)
```
Comment (arnavb) [inline Unique_Numbers/unique_number.cpp]: Pre-increment.

File: Unique_Numbers/unique_number.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+        cout << arr[i] << " "; 
+    }
+}
+
+int main()
+{
+    int n;
+    cout << "Enter the size of array : \n";
+    cin >> n;
+    int arr[n];
```
Comment (arnavb) [inline Unique_Numbers/unique_number.cpp]: This is not standards compliant C++. Use `std::vector<int>` instead.

File: Unique_Numbers/unique_number.cpp [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    cout << "Enter the values for the array : \n";
+    for(int i = 0; i < n; i++)
+        cin >> arr[i];
+    cout << "The unique elements are : \n";
+    display_unique(arr, n);
+
```
Comment (arnavb) [inline Unique_Numbers/unique_number.cpp]: Add a newline here.

**Comment by HariniJeyaraman** [comment]
@arnavb I have reformatted to Clang, used preincrement, used std::vector<int> instead of arr[n],  and corrected the file path. Please review my code.

**Comment by arnavb** [comment]
Last change, remove the using namespace std and use std:: everywhere.

**Comment by HariniJeyaraman** [comment]
@arnavb removed namespace std. Made all changes requested

**Comment by arnavb** [comment]
Sorry! I forgot to tell you to use 4 space indentation, but that is my last change (I know I'm saying this twice, sorry!). 

Here I reformatted it for you, you just have to copy paste it (I used https://zed0.co.uk/clang-format-configurator/):

```C++
#include <algorithm>
#include <iostream>
#include <stdlib.h>
#include <vector>

void display_unique(std::vector<int> arr, int n) {
    sort(arr.begin(), arr.end());
    for (int i = 0; i < n; i++) {
        while (i < n - 1 && arr[i] == arr[i + 1])
            i++;
        std::cout << arr[i] << " ";
    }
}

int main() {
    int n, val;
    std::cout << "Enter the size of array : \n";
    std::cin >> n;
    std::vector<int> arr;
    cout << "Enter the values for the array : \n";
    for (int i = 0; i < n; i++) {
        std::cin >> val;
        arr.push_back(val);
    }
    std::cout << "The unique elements are : \n";
    display_unique(arr, n);

    return 0;
}

/*Sample Input- Output
Enter the size of array :
5
Enter the values for the array :
12 34 15 12 16
The unique elements are :
12 15 16 34
*/
```

**Comment by HariniJeyaraman** [comment]
@arnavb made the last changes requested. Please review again.

**Comment by HariniJeyaraman** [comment]
@arnavb Can you please review this code ? Made all changes requested

File: code/unclassified/src/unique_number/unique_number.cpp [hunk 7]
```diff
@@ -0,0 +1,38 @@
+#include <algorithm>
+#include <iostream>
+#include <stdlib.h>
```
Comment (arnavb) [inline code/unclassified/src/unique_number/unique_number.cpp]: Use `cstdlib`

File: code/unclassified/src/unique_number/unique_number.cpp [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+#include <vector>
+
+void display_unique(std::vector<int> arr, int n) {
+    sort(arr.begin(), arr.end());
+    for (int i = 0; i < n; i++) {
```
Comment (arnavb) [inline code/unclassified/src/unique_number/unique_number.cpp]: Pre increment whenever in loop condition.

File: code/unclassified/src/unique_number/unique_number.cpp [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+int main() {
+    int n, val;
+    std::cout << "Enter the size of array : \n";
+    std::cin >> n;
+    std::vector<int> arr;
+    cout << "Enter the values for the array : \n";
+    for (int i = 0; i < n; i++) {
+        std::cin >> val;
+        arr.push_back(val);
+    }
```
Comment (arnavb) [inline code/unclassified/src/unique_number/unique_number.cpp]:
```suggestion
    std::vector<int> arr(n);
    cout << "Enter the values for the array : \n";
    for (int i = 0; i < n; i++) {
        std::cin >> val;
        arr[i] = val;
    }
```

**Pull request merged**